spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/core/Memoizer.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.spring.integration.core;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+/**
+ * Memorize function execution result
+ *
+ * @author Warren Zhu
+ */
+public class Memoizer {
```

Comment:
`Memoizer` typo? And the method names below too.

Reply:
I followed from https://opencredo.com/lambda-memoization-in-java-8/ and https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/concurrent/Memoizer.html